proposal/span.py

```diff
@@ -0,0 +1,27 @@
+from opentracing.span import Span as OTBaseSpan
+
+
+class Span(OTBaseSpan):
+    """Span class that provides some extensions to the Python OpenTracing API.
+    This class includes all methods that may be added to the OT Span class,
+    so that the proposed interface is public accessible
+    """
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Ends context manager and calls finish() on the span.
+        If exception has occurred during execution, it is automatically added
+        as a tag to the span.
+
+        Note: this method extends the current context manager
+        """
+        if exc_type:
+            self.log_kv({
+                'python.exception.type': exc_type,
+                'python.exception.val': exc_val,
+                'python.exception.tb': exc_tb,
+                })
+
+        # when the context manager is closed, we deactivate the current
+        # active Span before calling the finish()
+        self._tracer.active_span_source.deactivate(self)
+        self.finish()
```

Comment:
One interesting thing to consider is whether this should happen at `Span` level directly (right in the `opentracing` base code). *If* this is the taken path, we may need to add a field so such `Span` knows whether it should call `deactivate()` when `finish`ing (as we will likely have both `start_active_span` and `start_manual_span` from the Tracer.

Also, I think we should consider doing the deactivation from `finish()` itself, in case the user doesn't use the context manager, but simply calls `finish` by herself/himself.

Let me know what you think. As always, I can cook up samples as/if needed ;)